### ISSUE TITLE
freshrss service not working

### ISSUE THREAD

**Issue by bfrascher** [mother-of-all-self-hosting/mash-playbook#116]
First and foremost, thanks for this awesome collection of services.  It has saved me a lot of time and effort with setup and maintenance, as well as allowed me to discover new useful services to try out.

The only issue I have encountered so far is, that I can't get the freshrss service to work.  After setting it up (with the default values) I always get a 404 status code when accessing the "/freshrss" path on my server.  I noticed the following error message, printed multiple times, in the logs of the mash-freshrss systemd service:

`[unixd:alert] [pid 38] (1)Operation not permitted: AH02157: initgroups: unable to set groups for User www-data and Group 33`

I tried this with multiple versions of the freshrss docker image, but always with the same result.  Any help with this would be very much appreciated.

**Comment by spantaleev** [comment]
Thank you for the kind words! It's great to hear that the playbook is useful to you!

Perhaps @kinduff knows more about the current status of the FreshRSS service.

In the meantime, if you can dedicate a hostname to FreshRSS, you could try:

- adjusting `freshrss_hostname` to point to that hostname
- removing `freshrss_path_prefix: /freshrss` from your `vars.yml` file
- reinstalling
- checking if FreshRSS works on this hostname (without being hosted at a subpath). Perhaps subpath/prefix-hosting is experiencing issues for this service.

**Comment by kinduff** [comment]
Hello, I just gave it a try in a fresh server with the following configuration and is working as expected.

```yaml
freshrss_enabled: true
freshrss_hostname: freshrss.example.com
```

In the other hand, the `freshrss_path_prefix: /freshrss` configuration does not work as expected. I will work on a fix and update this issue.

In the meantime, please use a subdomain to serve the application.

**Comment by spantaleev** [comment]
Does FreshRSS ask you for the database password explicitly as some part of setup wizard?

If so, sounds like we need to update its installation docs and likely advise people to use an explicitly-specified password instead of the current one (auto-derived from the generic secret key).

You can switch to your own password by defining `freshrss_database_password: SOMETHING_SECRET` in your `vars.yml` file and re-running the installation (`just run-tags install-postgres,install-freshrss,start`). This will provoke the Postgres role to change the password for this database and will reconfigure FreshRSS to use the new password.

**Comment by bfrascher** [comment]
Yes, the setup wizard asks for which database to use and for postgres all connection options need to be entered explicitly (server, database name, user, password).

I have since set the password manually and was able to complete the setup, thanks.

**Comment by spantaleev** [comment]
In that case it seems like:

- `group_vars/mash_servers` should not try to auto-generate a password for FreshRSS (`freshrss_database_password`)
- validation tasks in the FreshRSS role should ensure that `freshrss_database_password` is non-empty
- `docs/services/freshrss.md` should require people to define `freshrss_database_password` in their `vars.yml` file

Validation tasks were already in place, so I've only had to tackle the other 2 things. I did that in https://github.com/mother-of-all-self-hosting/mash-playbook/commit/65fac6b4303a2bfb8f4adb2c1276887196869b34

-----

The only remaining things to be solved in this issue is that inability to use `freshrss_path_prefix`.

I've added a note about it in this patch until a proper fix arrives: https://github.com/mother-of-all-self-hosting/mash-playbook/commit/67f62f5d8a906aa28b78a7a428ab4337b5acd5e6

**Comment by kinduff** [comment]
Thank you so much for adding a patch @spantaleev, you beat me to it :stuck_out_tongue_closed_eyes: Will provide an update once the path issue is solved.

**Comment by luixxiul** [comment]
It's been fixed at some point.
